src/auth.ts

```diff
@@ -548,29 +553,73 @@ export class BasicAuth extends AuthType implements IBasicAuthType {
   }
 }
 
+/**
+ * hidden
+ */
 export abstract class NativeAuthType {
```

Comment:
I think this should just be `NativeAuth`. It is purposefully separate from the `Auth` class, which is why auth types exist (because of `login('facebook', ...` etc)